src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -346,4 +342,37 @@ public function get_collection_params() {
 		 */
 		return apply_filters( 'rest_pattern_directory_collection_params', $query_params );
 	}
+
+	/*
+	 * Include a hash of the query args, so that different requests are stored in
+	 * separate caches.
+	 *
+	 * MD5 is chosen for its speed, low-collision rate, universal availability, and to stay
+	 * under the character limit for `_site_transient_timeout_{...}` keys.
+	 *
+	 * @link https://stackoverflow.com/questions/3665247/fastest-hash-for-non-cryptographic-uses
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $query_args Query arguments to generate a transient key from.
+	 *
+	 * @return string Transient key.
+	 */
+	protected function get_transient_key( $query_args ) {
+
+		if ( isset( $query_args['slug'] ) ) {
+			// This is an additional precaution because the "sort" function expects an array.
+			$query_args['slug'] = wp_parse_list( $query_args['slug'] );
+
+			// Sort the array so that the transient key doesn't depend on the order of slugs.
+			sort( $query_args['slug'] );
+
+			// Empty arrays should not affect the transient key.
+			if ( 0 === count( $query_args['slug'] ) ) {
+				unset( $query_args['slug'] );
+			}
```

Comment:
Micro-optimization suggestions:

1. Only invoke `sort()` when the `array` has elements to sort.
2. Use `empty()` to detect that the array is empty. Why? (a) avoids the overhead with `count()`, (b) teeny tiny faster, (c) a wee bit more understandable, and (d) checks for a `false` return from `wp_parse_list()` (though highly unlikely a `false` will be returned.

```suggestion
			// Empty arrays should not affect the transient key.
			if ( empty( $query_args['slug'] ) ) {
				unset( $query_args['slug'] );
			} else {
				// Sort the array so that the transient key doesn't depend on the order of slugs.
				sort( $query_args['slug'] );
			}
```

See it in action https://3v4l.org/kOqb5.

Reply:
Resolved in https://github.com/WordPress/wordpress-develop/pull/2625/commits/1d8f893540d5fb7755444376d606059cfd677508.

Reply:
I do agree with your suggestions, @hellofromtonya.
Thank you!
I will allow myself to comment on them :) 
1. I hoped that the "sort" function would have a check not to sort empty arrays. But I'm not 100% sure. So, yes, let's not call it on empty arrays.
2. Well, this is news to me. I thought that the `count` function is faster because it knows the type of the input data( i.e., arrays), unlike the `empty` function, which does not initially know the type of data it is working with. I was wrong. That's good to know.